backend/src/AppServer.hs

```diff
@@ -174,9 +177,33 @@ getLeaderboardHandler year =
 adminRenamePlayerHandler :: RenamePlayerRequest -> AppM NoContent
 adminRenamePlayerHandler RenamePlayerRequest {pid, newName} = runDb $ updatePlayerName pid newName >> pure NoContent
 
+adminModifyReportHandler :: ModifyReportRequest -> AppM NoContent
+adminModifyReportHandler ModifyReportRequest {rid, report} = case validateReport report of
+  Failure errors -> throwError $ err422 {errBody = show errors}
+  Success _ -> runDb $ do
+    oldReport <- readOrError ("Cannot find report " <>: rid) $ lift . get $ rid
+    Entity newWinnerId _ <- readOrError ("Cannot find player " <>: report.winner) $ getPlayerByName report.winner
+    Entity newLoserId _ <- readOrError ("Cannot find player " <>: report.loser) $ getPlayerByName report.loser
```

Comment:
Idk how to handle the specific case of someone entering an **existing player** in the original report, then an admin wanting to reassociate the report with a **new player** that doesn't exist yet. Is that a realistic thing that ever happens?

The combination of APIs you just added would already let an admin work around this though which is sweet. Submit a dummy report with the new player, reassociate the target report to the new player, then delete the dummy report

Reply:
That is IMO an incredibly unlikely event - at the very least I've never seen it happen. I think not worth addressing yet (esp. because of your identified workaround). That said, this is a really nice catch of a weird corner case, and it's good to have it back-of-mind.